controllers/etcd_controller.go

```diff
@@ -588,6 +594,151 @@ const (
 	noOp        operationResult = "none"
 )
 
+func calculatePDBminAvailable(etcd *druidv1alpha1.Etcd) int32 {
+	// cluster is in bootstrap
+	if etcd.Status.ClusterSize == nil {
+		return 0
+	}
+
+	allMembersReady := false
+	backupReady := false
+	for _, condition := range etcd.Status.Conditions {
+		if condition.Type == druidv1alpha1.ConditionTypeAllMembersReady &&
+			condition.Status == druidv1alpha1.ConditionStatus(druidv1alpha1.EtcdMemberStatusReady) {
+			allMembersReady = true
+			continue
+		}
+
+		if condition.Type == druidv1alpha1.ConditionTypeBackupReady &&
+			condition.Status == druidv1alpha1.ConditionTrue {
+			backupReady = true
+			continue
+		}
```

Comment:
This is another use-case why the custodian_controller ought to patch the PDB as well. As soon as the condition change, the proper minAvailable value must be updated.